From cecccb018e80752597dbd8c0b3320125b0ec2cf0 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 4 Mar 2025 15:26:19 +0100
Subject: [PATCH] drm/panel: lvds: add delay after prepare

Delays are missing to the panel lvds. Some graphical effects can be
observed at the beginning of the display.

Change-Id: I94eb3e8117c5b637be6c1a568c01d4f0cede9efe
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/441603
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/panel/panel-lvds.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-lvds.c b/drivers/gpu/drm/panel/panel-lvds.c
index 54cd674f604c..adb35047efbf 100644
--- a/drivers/gpu/drm/panel/panel-lvds.c
+++ b/drivers/gpu/drm/panel/panel-lvds.c
@@ -8,6 +8,7 @@
  * Contact: Laurent Pinchart (laurent.pinchart@ideasonboard.com)
  */
 
+#include <linux/delay.h>
 #include <linux/gpio/consumer.h>
 #include <linux/module.h>
 #include <linux/of.h>
@@ -87,6 +88,13 @@ static int panel_lvds_prepare(struct drm_panel *panel)
 	if (lvds->enable_gpio)
 		gpiod_set_value_cansleep(lvds->enable_gpio, 1);
 
+	/*
+	 * ETML070016NDHA LVDS panel requires at least 16ms and several VSD cycles
+	 * between enable regulators and first frame. The delay sets by defaults to 30ms,
+	 * device tree needs to be updated to add new properties for delays.
+	 */
+	msleep(30);
+
 	lvds->prepared = true;
 
 	return 0;
-- 
2.50.1

